### ISSUE TITLE
Improve log observability of backend. 

### ISSUE THREAD

**Issue by zhuyuanmao** [hyperledger-cello/cello#471]
While we were developing the backend of the Cello project, we didn't put much effort into structuring log information, which would be a painful technical debit in terms of maintainability and testability. We want to refine our log events for every error handling.  Ultimately, we will achieve good observabilities and will be beneficial for long-term development.   There is an example of PR we did for this task.  #465

**Comment by syedmustafan** [comment]
### The logs should be maintained.

Should create a new model for the logs. All the logs should be maintained in the table.

**Comment by zhuyuanmao** [comment]
> #### The logs should be maintained.
> Should create a new model for the logs. All the logs should be maintained in the table.

good point, but we don't have a plan to maintain the logs at this moment.

**Comment by Survey845** [comment]
Hi @zhuyuanmao, can I take up this issue in case no one is working on it?

**Comment by Survey845** [comment]
I went through the PR mentioned in the original post and wanted to ask if I should be looking for code where exceptions are raised and then add some text to output what is not working. For example, the following is a code snippet from handler.py ( inside ./src/api-engine/api/lib/agent):

```python
def create(self, info):
        try:
            cid = self._agent.create(info)
            if cid:
                return cid
            else:
                return None
        except Exception as e:
            raise e
```

Should I use LOG.error() before raising the exception for better observability?:
```python
except Exception as e:
    LOG.error("Text about what went wrong here", exc_info=True) 
    raise e
```


Or should I follow: 

```python
except Exception as e:
    raise Exception("Text about what went wrong here") from e
```

**Comment by yeasy** [comment]
The first method looks better? @xichen1 @YoungHypo

**Comment by YoungHypo** [comment]
The first is better. I think the second one wraps different types of errors into a generic Exception, which results in the loss of the original exception type. @yeasy @Survey845

**Comment by Survey845** [comment]
Alright then, will move forward with the first option.@YoungHypo  @yeasy

**Comment by Survey845** [comment]
Sorry for not mentioning it earlier, there are a few more files where I can add logging, the PR just had my first commit, and I wanted to know if I should use a similar process for all other files. Let me know if making another PR with all the changes combined will suffice or does the issue itself have to be reopened? @YoungHypo @yeasy
